mcda/utils/utils_for_main.py

```diff
@@ -0,0 +1,1200 @@
+    - UserStoppedInfo: If the user chooses to stop when duplicates are found.
+
+     :rtype: pd.DataFrame
+     :return: input_matrix
+     :param input_matrix: pd.DataFrame
+    """
+    if input_matrix.duplicated().any():
+        raise ValueError('Error: Duplicated rows in the alternatives column.')
+    elif input_matrix.iloc[:, 0].duplicated().any():
+        logger.info('Duplicated rows in the alternatives column.')
+
+    index_column_values = input_matrix.index.tolist()
+    logger.info("Alternatives are {}".format(index_column_values))
+    input_matrix_no_alternatives = input_matrix.reset_index(drop=True)  # drop the alternative
+
+    input_matrix_no_alternatives = _check_and_rescale_negative_indicators(
+        input_matrix_no_alternatives)
+
+    return input_matrix_no_alternatives
+
+
+def read_matrix(input_matrix_path: str) -> pd.DataFrame():
+    """
+    Read an input matrix from a CSV file and return it as a DataFrame.
+    Set the 'Alternatives' column as index column.
+
+    Parameters:
+    - input_matrix_path (str): Path to the CSV file containing the input matrix.
+
+    Raises:
+    - Exception: If an error occurs during the file reading or DataFrame creation.
+
+    :rtype: pd.DataFrame
+    :param input_matrix_path: str
+    """
+
+    try:
+        filename = input_matrix_path
+        with open(filename, 'r') as fp:
+            matrix = pd.read_csv(fp, sep="[,;:]", decimal='.', engine='python')
+            data_types = {col: 'float64' for col in matrix.columns[1:]}
+            matrix = matrix.astype(data_types)
+            alternatives_column_name = matrix.columns[0]
+            matrix = matrix.set_index(alternatives_column_name)
+            return matrix
+    except Exception as e:
+        print(e)
+
+
+def reset_index_if_needed(series):
+    """
+    Reset the index of a pandas Series if it's not a RangeIndex.
+
+    Parameters:
+    - series (pd.Series): The input pandas Series.
+
+    Returns:
+    - pd.Series: The series with the index reset if needed.
+
+    # Usage example:
+    res = reset_index_if_needed(res)
+    """
+    if not isinstance(series.index, pd.RangeIndex):
+        series = series.reset_index(drop=True)
+    return series
+
+
+def _check_and_rescale_negative_indicators(input_matrix: pd.DataFrame) -> pd.DataFrame():
+    """If some indicators in the input matrix are negative, they are rescaled into [0-1]"""
+
+    if (input_matrix < 0).any().any():
+        scaler = MinMaxScaler()
+        scaled_data = scaler.fit_transform(input_matrix)
+        scaled_matrix = pd.DataFrame(
+            scaled_data, columns=input_matrix.columns, index=input_matrix.index)
+        return scaled_matrix
+    else:
+        return input_matrix
+
+
+def parse_args():
+    """
+    Parse command line arguments for configuration path.
+
+    Raises:
+    - argparse.ArgumentError: If the required argument is not provided.
+
+    :rtype: str
```

Comment:
I believe `:rtype` is to defined the return type of the method. Either it should be present for all the methods or none. Please modify as per the documentation.